[Feature] jemalloc lib for `x86_64-pc-windows-msvc` so can't install snarkOS with VSC (Visual Studio Code) on Windows

## 🚀 Feature

It's not a new one but would be awesome to have the option of installing snarkOS with Visual Studio Code.
Tried manually installing all needed jemalloc libs, but nothing helped.

## Motivation

A lot of Windows users here.

## Full log here
```
warning: "jemalloc support for `x86_64-pc-windows-msvc` is untested"

error: failed to run custom build command for `tikv-jemalloc-sys v0.5.3+5.3.0-patched`

Caused by:
  process didn't exit successfully: `C:\Users\user\snarkOS\target\release\build\tikv-jemalloc-sys-0341649976ece1cf\build-script-build` (exit code: 101)
  --- stdout
  TARGET=x86_64-pc-windows-msvc
  HOST=x86_64-pc-windows-msvc
  NUM_JOBS=8
  OUT_DIR="C:\\Users\\user\\snarkOS\\target\\release\\build\\tikv-jemalloc-sys-99d1ae042f58a7ea\\out"
  BUILD_DIR="C:\\Users\\user\\snarkOS\\target\\release\\build\\tikv-jemalloc-sys-99d1ae042f58a7ea\\out\\build"
  SRC_DIR="C:\\Users\\user\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\tikv-jemalloc-sys-0.5.3+5.3.0-patched"
  cargo:warning="jemalloc support for `x86_64-pc-windows-msvc` is untested"
  cargo:rustc-cfg=prefixed
  cargo:rerun-if-env-changed=JEMALLOC_OVERRIDE
  OPT_LEVEL = Some("3")
  TARGET = Some("x86_64-pc-windows-msvc")
  HOST = Some("x86_64-pc-windows-msvc")
  cargo:rerun-if-env-changed=CC_x86_64-pc-windows-msvc
  CC_x86_64-pc-windows-msvc = None
  cargo:rerun-if-env-changed=CC_x86_64_pc_windows_msvc
  CC_x86_64_pc_windows_msvc = None
  cargo:rerun-if-env-changed=HOST_CC
  HOST_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CFLAGS_x86_64-pc-windows-msvc
  CFLAGS_x86_64-pc-windows-msvc = None
  cargo:rerun-if-env-changed=CFLAGS_x86_64_pc_windows_msvc
  CFLAGS_x86_64_pc_windows_msvc = None
  cargo:rerun-if-env-changed=HOST_CFLAGS
  HOST_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  DEBUG = Some("false")
  CC="C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX64\\x64\\cl.exe"
  CFLAGS="-nologo -MD -O2 -Brepro -W4"
  JEMALLOC_REPO_DIR="jemalloc"
  cargo:rerun-if-env-changed=JEMALLOC_SYS_WITH_MALLOC_CONF
  cargo:rerun-if-env-changed=JEMALLOC_SYS_WITH_LG_PAGE
  cargo:rerun-if-env-changed=JEMALLOC_SYS_WITH_LG_HUGEPAGE
  cargo:rerun-if-env-changed=JEMALLOC_SYS_WITH_LG_QUANTUM
  cargo:rerun-if-env-changed=JEMALLOC_SYS_WITH_LG_VADDR
  --with-jemalloc-prefix=_rjem_
  running: "sh" "/c/Users/user/snarkOS/target/release/build/tikv-jemalloc-sys-99d1ae042f58a7ea/out/build/configure" "--disable-cxx" "--enable-doc=no" "--enable-shared=no" "--with-jemalloc-prefix=_rjem_" "--with-private-namespace=_rjem_" "--host=x86_64-pc-win32" "--build=x86_64-pc-win32" "--prefix=C:\\Users\\user\\snarkOS\\target\\release\\build\\tikv-jemalloc-sys-99d1ae042f58a7ea\\out"
  checking for xsltproc... false
  checking for x86_64-pc-win32-gcc... C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\cl.exe
  checking whether the C compiler works... no
  running: "tail" "-n" "100" "C:\\Users\\user\\snarkOS\\target\\release\\build\\tikv-jemalloc-sys-99d1ae042f58a7ea\\out\\build\\config.log"
  XSLROOT=''
  XSLTPROC='false'
  a=''
  abi=''
  abs_objroot='/c/Users/user/snarkOS/target/release/build/tikv-jemalloc-sys-99d1ae042f58a7ea/out/build/'
  abs_srcroot='/c/Users/user/snarkOS/target/release/build/tikv-jemalloc-sys-99d1ae042f58a7ea/out/build/'
  ac_ct_CC=''
  ac_ct_CXX=''
  bindir='${exec_prefix}/bin'
  build='x86_64-pc-win32'
  build_alias='x86_64-pc-win32'
  build_cpu=''
  build_os=''
  build_vendor=''
  cfghdrs_in=''
  cfghdrs_out=''
  cfgoutputs_in=''
  cfgoutputs_out=''
  datadir='${datarootdir}'
  datarootdir='${prefix}/share'
  docdir='${datarootdir}/doc/${PACKAGE}'
  dvidir='${docdir}'
  enable_autogen=''
  enable_cache_oblivious=''
  enable_cxx='no'
  enable_debug=''
  enable_doc='no'
  enable_experimental_smallocx=''
  enable_fill=''
  enable_initial_exec_tls=''
  enable_lazy_lock=''
  enable_log=''
  enable_opt_safety_checks=''
  enable_opt_size_checks=''
  enable_prof=''
  enable_readlinkat=''
  enable_shared='no'
  enable_static=''
  enable_stats=''
  enable_tls=''
  enable_uaf_detection=''
  enable_utrace=''
  enable_xmalloc=''
  enable_zone_allocator=''
  exe=''
  exec_prefix='C:\Users\user\snarkOS\target\release\build\tikv-jemalloc-sys-99d1ae042f58a7ea\out'
  host='x86_64-pc-win32'
  host_alias='x86_64-pc-win32'
  host_cpu=''
  host_os=''
  host_vendor=''
  htmldir='${docdir}'
  importlib=''
  includedir='${prefix}/include'
  infodir='${datarootdir}/info'
  install_suffix=''
  je_=''
  jemalloc_version=''
  jemalloc_version_bugfix=''
  jemalloc_version_gid=''
  jemalloc_version_major=''
  jemalloc_version_minor=''
  jemalloc_version_nrev=''
  libdir='${exec_prefix}/lib'
  libdl=''
  libexecdir='${exec_prefix}/libexec'
  libprefix=''
  link_whole_archive=''
  localedir='${datarootdir}/locale'
  localstatedir='${prefix}/var'
  mandir='${datarootdir}/man'
  o=''
  objroot=''
  oldincludedir='/usr/include'
  pdfdir='${docdir}'
  prefix='C:\Users\user\snarkOS\target\release\build\tikv-jemalloc-sys-99d1ae042f58a7ea\out'
  private_namespace=''
  program_transform_name='s,x,x,'
  psdir='${docdir}'
  rev='2'
  sbindir='${exec_prefix}/sbin'
  sharedstatedir='${prefix}/com'
  so=''
  srcroot=''
  sysconfdir='${prefix}/etc'
  target_alias=''

  ## ----------- ##
  ## confdefs.h. ##
  ## ----------- ##

  /* confdefs.h */
  #define PACKAGE_NAME ""
  #define PACKAGE_TARNAME ""
  #define PACKAGE_VERSION ""
  #define PACKAGE_STRING ""
  #define PACKAGE_BUGREPORT ""
  #define PACKAGE_URL ""

  configure: exit 77

  --- stderr
  configure: error: in `/c/Users/user/snarkOS/target/release/build/tikv-jemalloc-sys-99d1ae042f58a7ea/out/build':
  configure: error: C compiler cannot create executables
  See `config.log' for more details
  thread 'main' panicked at 'command did not execute successfully: "sh" "/c/Users/user/snarkOS/target/release/build/tikv-jemalloc-sys-99d1ae042f58a7ea/out/build/configure" "--disable-cxx" "--enable-doc=no" "--enable-shared=no" "--with-jemalloc-prefix=_rjem_" "--with-private-namespace=_rjem_" "--host=x86_64-pc-win32" "--build=x86_64-pc-win32" "--prefix=C:\\Users\\user\\snarkOS\\target\\release\\build\\tikv-jemalloc-sys-99d1ae042f58a7ea\\out"
  expected success, got: exit code: 77', C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\tikv-jemalloc-sys-0.5.3+5.3.0-patched\build.rs:346:9
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: failed to compile `snarkos v2.0.2 (C:\Users\user\snarkOS)`, intermediate artifacts can be found at `C:\Users\user\snarkOS\target`
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -37,6 +37,10 @@ members = [
 name = "snarkos"
 path = "snarkos/main.rs"
 
+[features]
+default = [ "jemalloc" ]
+jemalloc = [ "tikv-jemallocator" ]
+
 [workspace.dependencies.snarkvm]
 #path = "../snarkVM"
 #git = "https://github.com/AleoHQ/snarkVM.git"
@@ -89,6 +93,7 @@ path = "./node/tcp"
 
 [dependencies.tikv-jemallocator]
 version = "0.5"
+optional = true
 
 [dev-dependencies.rusty-hook]
 version = "0.11.2"

--- a/snarkos/main.rs
+++ b/snarkos/main.rs
@@ -18,8 +18,10 @@ use snarkos_cli::{commands::CLI, helpers::Updater};
 use snarkos_node_env::EnvInfo;
 
 use clap::Parser;
+#[cfg(feature = "jemalloc")]
 use tikv_jemallocator::Jemalloc;
 
+#[cfg(feature = "jemalloc")]
 #[global_allocator]
 static GLOBAL: Jemalloc = Jemalloc;
 

